io_scene_godot/converters/mesh.py

```diff
@@ -25,19 +26,23 @@ def export_mesh_node(escn_file, export_settings, obj, parent_gd_node):
         parent_gd_node = physics.export_physics_properties(
             escn_file, export_settings, obj, parent_gd_node
         )
-
-    if physics.has_physics(obj) and obj.display_type == "WIRE":
         # skip wire mesh which is used as collision mesh
-        return parent_gd_node
+        if obj.display_type == "WIRE":
+            return parent_gd_node
 
     mesh_node = NodeTemplate(obj.name, "MeshInstance", parent_gd_node)
-    mesh_exporter = MeshResourceExporter(obj)
+    mesh_exporter = ArrayMeshResourceExporter(obj)
 
     armature_data = get_modifier_armature_data(obj)
     if ("ARMATURE" in export_settings['object_types'] and
             armature_data is not None):
+        # set armature to REST before exporting meshes
         armature_pose_position = armature_data.pose_position
         armature_data.pose_position = "REST"
+        # TODO: Not correct to assume the armature is closest parent. Armature
```

Comment:
Great! You found this issue as well lol. It has been brought up in #194, I thought set a NodePath after exporting all nodes may not be enough.  Because in some case, the mesh needs to know information of its skeleton to be able to export vertex weights (e.g. when IK control bones are configured to not exported). 

IMO, we may need two traversals on blender scene tree. The first pass to built the node relations and the second pass export actual data.

Reply:
Agreed. But is there any other reason to do 2 traversals besides skeletons? Just curious.

Firstly, I suggest we add a bl_object_gd_node_map to ESCNFile, to which we add our NodeTemplate objects as values and the blender objects they're based on as keys (during export_object).

At the end of export_scene() we look through the exported blender objects, find the pairs of mesh objects and armatures that need to be linked, find the godot nodes with the map and set the node["skeleton"] links here.

Reply:
By saying two passes,  I was mean to solve all the 'forward dependencies', not limit to Skeleton-Mesh. And I was thinking it would not be easy to fetch information of `exporting_bones` while exporting the mesh.  Well, by looking at your code, I found it's actually very easy to it :D 

Your approach is very good and certainly works, I would adopt it. Thanks